src/game/stage/game_7.c

```diff
@@ -84,11 +84,18 @@ void sub_802DBC0(u8 p0, u16 p1)
 // (84.45%) https://decomp.me/scratch/ge5fv
 NONMATCH("asm/non_matching/game/stage/sub_802DCC8.inc", void sub_802DCC8(u8 p0, u16 p1))
 {
+    s32 r2;
+#ifndef NON_MATCHING
     register u32 r7 asm("r7") = p0;
     u8 *bgOffsets = gBgOffsetsHBlank;
-    s32 r2;
     register s32 r5 asm("r5") = 0;
     register u16 r6 asm("r6") = ((unsigned)p1 << 22) >> 22;
+#else
+    u32 r7 = p0;
+    u8 *bgOffsets = gBgOffsetsHBlank;
+    s32 r5 = 0;
+    u16 r6 = ((unsigned)p1 << 22) >> 22;
```

Comment:
This definitely isn't the matching code, couldn't you just write `r6 = p1`? This to me looks like p1 could be a u8 though